packages/react-scripts/scripts/build.js

```diff
@@ -112,7 +112,7 @@ function build(previousFileSizes) {
       if (messages.errors.length) {
         return reject(new Error(messages.errors.join('\n\n')));
       }
-      if (process.env.CI && messages.warnings.length) {
+      if (process.env.CI && process.env.CI.toLowerCase() === 'true' && messages.warnings.length) {
```

Comment:
Given the way it works currently (truthy), using !== false would retain current functionality and wouldn't be a breaking change (if you consider 'false' a bug).

Even if 'true' is the proper value to evaluate against, we would have to postpone that to 2.0.0.